awstesting/util.go

```diff
@@ -95,13 +95,30 @@ func (c *FakeContext) Value(key interface{}) interface{} {
 	return nil
 }
 
-// StashEnv stashes the current environment variables and returns an array of
-// all environment values as key=val strings.
-func StashEnv() []string {
-	env := os.Environ()
-	os.Clearenv()
+// StashEnv stashes the current environment variables except variables listed in envToKeep
+// Returns an array of all environment values as key=val strings.
+func StashEnv(envToKeep ...string) []string {
```

Comment:
Suggest updating `aws/credentials/processcreds` test, `preserveImportantStashEnv` function to use this updated utility.